nautobot/extras/models/models.py

```diff
@@ -667,23 +670,23 @@ class Notes(BaseModel, ChangeLoggedModel):
         null=True,
     )
     user_name = models.CharField(max_length=150, editable=False)
-    name = models.CharField(max_length=150)
-    slug = AutoSlugField(populate_from="name", max_length=150)
+
+    slug = AutoSlugField(populate_from="assigned_object", max_length=150)
     note = models.TextField()
     objects = NotesQuerySet.as_manager()
 
     class Meta:
         ordering = ["created"]
-        verbose_name = "Note"
-        verbose_name_plural = "Notes"
+
+    def slugify_function(self, content):
+        return slugify(f"{content}-{datetime.now().isoformat()}")
```

Comment:
For safety, should truncate `content` to an appropriate maximum length, as some models (Job in particular, but also thinking of potential plugin models) can have very long string representations.

```suggestion
        return slugify(f"{content[:50]}-{datetime.now().isoformat()}")
```